elevate-project/controllers/project.js

```diff
@@ -1,29 +1,75 @@
+const { transform } = require('lodash')
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
 const requestParser = require('../utils/requestParser')
 
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = {result: {data: [],count: 0}}
+	let proceed = true
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+	// construct body for fetching project templates 
+	let body = {
+		"query": {
+			"status": "published"
+		},
+		"projection": ["_id" , "title","createdBy","createdAt","description"],
+		"limit" : max_limit
+	}
+	// custom header 
+	let header = {
+		'internal-access-token' : req.headers['internal_access_token'],
+		'Content-Type' : 'application/json'
+	}
+	// enrich body if needed 
+	if(req.body){
+		if(req.body.search) body.query.title = {
+			"$regex" : req.body.search,
+			"$options" : 'i'
+		}
+
+		// check if any specific resource type is requested 
+		if(req?.body?.resourceType !== undefined || req?.body?.resourceType != []){
```

Comment:
@adithyadinesh0412 we can optimize 
 const resourceType = req.body?.resourceType || []; then u can check length of resourceType